lib/Driver/ToolChains.cpp

```diff
@@ -1211,12 +1211,33 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(LibProfile));
   }
 
-  // FIXME: We probably shouldn't be adding an rpath here unless we know ahead
-  // of time the standard library won't be copied.
-  Arguments.push_back("-Xlinker");
-  Arguments.push_back("-rpath");
-  Arguments.push_back("-Xlinker");
-  Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+  if (getTriple().isAndroid()) {
+    // FIXME: These should be set in CMake.
+    Arguments.push_back("-target");
+    Arguments.push_back("armv7-none-linux-androideabi");
+
+    const char* ndkhome = getenv("ANDROID_NDK_HOME");
+    assert(ndkhome && "ANDROID_NDK_HOME needs to be set to NDK "
+      "install directory for linking");
+
+    auto libgccpath = Twine(ndkhome) + "/toolchains/"
+      "arm-linux-androideabi-4.8/prebuilt/linux-x86_64/"
+      "lib/gcc/arm-linux-androideabi/4.8";
+    Arguments.push_back("-L");
+    Arguments.push_back(context.Args.MakeArgString(libgccpath));
+
+    auto libcxxpath = Twine(ndkhome) + "/sources/"
+      "cxx-stl/llvm-libc++/libs/armeabi-v7a";
+    Arguments.push_back("-L");
+    Arguments.push_back(context.Args.MakeArgString(libcxxpath));
+  } else {
+    // FIXME: We probably shouldn't be adding an rpath here unless we know ahead
+    // of time the standard library won't be copied.
+    Arguments.push_back("-Xlinker");
+    Arguments.push_back("-rpath");
+    Arguments.push_back("-Xlinker");
+    Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
```

Comment:
Typically, clang uses `-sysroot` to specify the path to a copy of the target filesystem.  But, I'd recommend looking into what clang actually does for Android, what is the recommended way to invoke it, and trying to replicate that, as long as it makes sense.